### PR TITLE
allow user to choose identity provider on signin

### LINKED ISSUE #168: Update Users feature to use WebIds instead of username
**Is your feature request related to a problem? Please describe.**
Currently, the Users feature works using by tying usernames (timbot, jfelt, hunter2) to User information created by the case manager (first name, last name, etc). There is also logic to use these usernames to access the user's pod and interact with them. However, usernames are not what are used in Solid to identify pods owned by an individual; Solid uses WebIds (https://timbot.solidcommunity.net/profile/card#me, https://www.timsBlog/solidPod/card#me) to identify individuals.

The code for the users feature currently attempts to take the username and interpolate it into a WebId when it tries to access a user's pod. However, this only works if the user's real WebId is in the format the code expects. There's no requirement for WebId formats beyond that they be a valid URI. If the WebId is actually a different format, the code will fail. 

**Describe the solution you'd like**
We can avoid a lot of these issues by storing the user's WebId instead of a username.

**Describe alternatives you've considered**
The WebId is harder to input than a simple username, however this can be mitigated with some clever UI.
We can auto-add the protocol (`https://`) so the case manager doesn't need to put that in. Additionally, we could provide templates for the WebId formats of the most popular OIDC providers (the current template works in probably 90% of cases, but we'd like the system to work in 100% of cases)

**Additional context**
Currently, we're also storing some user information in the user system. However, the WebId is intended to point to a public`profile` document. It's possible that instead of storing a large amount of info on each individual user, PASS could fetch the user's personal profile and display it to the case manager. This could reduce the amount of information we'd need to store in the PASS system.

### PR DESCRIPTION
**Pull request by timbot1789**
Adds an input field to PASS login that allows users to sign into a different identity provider if they so desire. By default, it loads with the `SOLID_IDENTITY_PROVIDER` configured in the enviroment. Also updates the login refresh logic to handle the new provider.

image

image

### CODE REVIEW

File: src/components/NavBar/NavBar.jsx
Comment (timbot1789) [inline src/components/NavBar/NavBar.jsx:60]: Luckily for us, this `removeKeys` function is built into browsers, so we don't need to build it ourselves.
`localStorage.clear()` removes ALL data we've put in localStorage, which will work fine as long as we use `localStorage` just for info that lets us preserve login between sessions, which we're doing now. If we need to store info across different active logins we have a few options:

1. Store it clientside in [indexeddb](https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API)
2. Store it serverside in a separate config file that we fetch on login.
3. Brainstorm a few other things too

These all have different advantages and tradeoffs.

Comment (leekahung) [reply src/components/NavBar/NavBar.jsx:60]: I had thought about a full clear before with localStorage.clear() before, but I've headed towards just clearing those generated by PASS.

**Comment by leekahung** [comment]
Not exactly the same as PR #169/Issue #168, but related. Linking issue #168 to this PR.

**Comment by leekahung** [comment]
This is a really small PR, but I'm going to hold this for now. Will approve after PR #157 is completed.

File: package.json
Comment (timbot1789) [inline package.json:56]: Add some libraries to test behavior of react components. React Testing Library is the one recommended by React MUI. React Testing Library runs code on the DOM level. I.E. it renders everything down to the DOM document sent to users, completely removing react, and making tests interact directly with HTML elements.

The other UI testing library I've used is enzyme, which keeps things at the react component level. Tests can then interact more easily with the React components themselves.

IMO, React Testing Library takes a better approach, since it forces the tests closer to what the user will actually see.

File: package.json
Comment (timbot1789) [inline package.json:57]: A recommended utility library on top of React Testing Library. Provides a lot of functions that simulate how a user actually interacts with a system. E.G. `user.type(input,'hello')` simulates the user mousing over the `input` element, clicking on it, then typing h-e-l-l-o on their keyboard.

File: test/components/NavBar/OidcLoginComponent.test.jsx
Comment (timbot1789) [inline test/components/NavBar/OidcLoginComponent.test.jsx:19]: Mock the Inrupt Login button so it doesn't try to fetch a session that doesn't exist. We don't care to test Inrupt's code here.

File: test/components/NavBar/OidcLoginComponent.test.jsx
Comment (timbot1789) [inline test/components/NavBar/OidcLoginComponent.test.jsx:9]: `@testing-library/react` adds all the `render` calls to a single shared document object. Without this call, renders between tests leak, which is bad.

File: src/components/NavBar/OidcLoginComponent.jsx
Comment (leekahung) [inline src/components/NavBar/OidcLoginComponent.jsx]: I believe these Boxes are here to help keep the components for TextField and the LoginButton on the right.

I think 1 Box should be enough since they're already using flexGrow: 1. The first box should fill out the space.

File: src/components/NavBar/OidcLoginComponent.jsx
Comment (leekahung) [inline src/components/NavBar/OidcLoginComponent.jsx:18]: TextField is off centered vertically, but that's a minor thing. Think we can update that in a future PR for styling.

**Review by leekahung — CHANGES_REQUESTED** [review]
Think this looks pretty good. Just a small change. See feedback.

File: src/components/NavBar/OidcLoginComponent.jsx
Comment (timbot1789) [reply src/components/NavBar/OidcLoginComponent.jsx]: Fixed

File: src/components/NavBar/OidcLoginComponent.jsx
Comment (timbot1789) [reply src/components/NavBar/OidcLoginComponent.jsx:18]: Fixed

**Review by leekahung — APPROVED** [review]
Alright, this looks good to me.

**Pull request merged**
